proxy/fadeintesting_test.go

```diff
@@ -107,18 +109,33 @@ func createDataClient(r ...*eskip.Route) fadeInDataClient {
 	var c fadeInDataClient
 	c.reset = make(chan []*eskip.Route, 1)
 	c.update = make(chan []*eskip.Route, 1)
+	c.quit = make(chan struct{})
 	c.reset <- r
 	return c
 }
 
 func (c fadeInDataClient) LoadAll() ([]*eskip.Route, error) {
-	r := <-c.reset
-	c.reset <- r
-	return r, nil
+	select {
+	case r := <-c.reset:
+		c.reset <- r
+		return r, nil
+	case <-c.quit:
+		return nil, nil
+	}
 }
 
 func (c fadeInDataClient) LoadUpdate() ([]*eskip.Route, []string, error) {
-	return <-c.update, nil, nil
+	select {
+	// hm, blocking dataclient?
```

Comment:
If you have 2 dataclients in use, what if only one fails such that 50% of your routes are not there. Do you want to startup normally or block and postpone/fail to start?
I think that's why we want to block in general.

Reply:
Its a bit weird as we have polling cycle per dataclient but it works both ways